src/IQFeed.CSharpApiClient/Socket/SocketClient.cs

```diff
@@ -111,7 +97,10 @@ void onCompleted(object sender, SocketAsyncEventArgs eventArgs)
             }
         }
 
-        public void Disconnect() { Dispose(); }
+        public void Disconnect()
+        {
+            DisconnectAsync().Wait();

```

Comment:
Replace synchronization of tasks like this:
`DisconnectAsync().GetAwaiter().GetResult();`
see https://stackoverflow.com/questions/36426937/what-is-the-difference-between-wait-vs-getawaiter-getresult/36427080#36427080